src/utils/props.js

```diff
@@ -104,13 +104,13 @@ export function getPropsData(element, componentDefinition, props) {
   const propsData = componentDefinition.propsData || {};
 
   props.hyphenate.forEach((name, index) => {
-    const elementAttribute = element.attributes[name];
     const propCamelCase = props.camelCase[index];
+    const propValue = element.attributes[name] || element[propCamelCase];
```

Comment:
Here's the root of the change. If the data was passed to the custom element by property ('.prop' modifier in Vue), then it will not be present in the DOM, but instead will be on the element at `element[propCamelCase]`.